mount/mount_linux.go

```diff
@@ -363,24 +364,29 @@ func mountAt(chdir string, source, target, fstype string, flags uintptr, data st
 		return unix.Mount(source, target, fstype, flags, data)
 	}
 
-	f, err := os.Open(chdir)
-	if err != nil {
-		return fmt.Errorf("failed to mountat: %w", err)
-	}
-	defer f.Close()
+	ch := make(chan error, 1)
+	go func() {
+		runtime.LockOSThread()
 
-	fs, err := f.Stat()
-	if err != nil {
-		return fmt.Errorf("failed to mountat: %w", err)
-	}
+		// Do not unlock this thread.
+		// If the thread is unlocked go will try to use it for other goroutines.
+		// However it is not possible to restore the thread state after CLONE_FS.
+		//
+		// Once the goroutine exits the thread should eventually be terminated by go.
```

Comment:
https://pkg.go.dev/runtime#LockOSThread

> If the calling goroutine exits without unlocking the thread, the thread will be terminated. 

So it will be fine.